src/services/ui/src/components/ExportButton/index.tsx

```diff
@@ -1,101 +1,88 @@
-      result[k] = format(new Date(value), "MM/dd/yyyy");
-    } else if (typeof value === "string" && key === "cmsStatus") {
-      if (isCms) {
-        result["Status"] = value;
-      }
-    } else if (typeof value === "string" && key === "stateStatus") {
-      if (!isCms) {
-        result["Status"] = value;
+
+    for (const item of resolvedData) {
+      const column: Record<any, any> = {};
+
+      for (const header of headers) {
+        column[header.name] = header.transform(item);
       }
-    } else {
-      result[k] = value;
+      exportData.push(column);
     }
-  }
 
-  return result;
-}
+    setLoading(false);
 
-export const OsExportButton = () => {
-  const [loading, setLoading] = useState(false);
-  const { data: user } = useGetUser();
-  const params = useOsParams();
+    return exportData;
+  };
 
-  const handleExport = async () => {
+  const handleExport = (data: Record<any, any>) => {
     const csvExporter = new ExportToCsv({
       useKeysAsHeaders: true,
       filename: `${params.state.tab}-export-${format(
         new Date(),
         "MM/dd/yyyy"
       )}`,
     });
-    setLoading(true);
-
-    const filters = DEFAULT_FILTERS[params.state.tab]?.filters ?? [];
 
-    const searchFilter = createSearchFilterable(params.state.search);
-    const osData = await getAllSearchData([
-      ...params.state.filters,
-      ...filters,
-      ...searchFilter,
-    ]);
-
-    const sourceItems = osData?.map((hit) => {
-      const filteredHit = formatDataForExport({ ...hit._source }, user?.isCms);
-
-      // Properties to exclude from export
-      Reflect.deleteProperty(filteredHit, "Attachments");
-      Reflect.deleteProperty(filteredHit, "Rai Responses");
-
-      return filteredHit;
-    });
-    csvExporter.generateCsv(sourceItems);
-
-    setLoading(false);
+    csvExporter.generateCsv(data);
   };
 
   return (
-    <Button
-      variant="ghost"
-      onClick={handleExport}
-      disabled={loading}
-      className="hover:bg-transparent h-full flex gap-2"
-    >
-      {loading && (
-        <motion.div
-          animate={{ rotate: "360deg" }}
-          transition={{ repeat: Infinity, duration: 0.5 }}
-        >
-          <Loader className="w-4 h-4" />
-        </motion.div>
-      )}
-      {!loading && <Download className="w-4 h-4" />}
-      <p className="prose-sm">Export</p>
-    </Button>
+    <>
+      <Button
+        variant="ghost"
+        onClick={async () => {
+          handleExport(await generateExport());
+        }}
+        disabled={loading}
+        className="hover:bg-transparent h-full flex gap-2"
+      >
+        {loading && (
+          <motion.div
+            animate={{ rotate: "360deg" }}
+            transition={{ repeat: Infinity, duration: 0.5 }}
+          >
+            <Loader className="w-4 h-4" />
+          </motion.div>
+        )}
+        {!loading && <Download className="w-4 h-4" />}
+        <p className="prose-sm">Export</p>
```

Comment:
Just a nitpick: I think this can be swapped to a span to adhere to HTML norms
https://stackoverflow.com/questions/26002381/p-tag-in-button-tag-not-allowed